ci/check-params-env.sh

```diff
@@ -407,66 +460,75 @@ function check_image() {
+    echo "ERROR: Variable names in the '${PARAMS_ENV_PATH}' & '${PARAMS_LATEST_ENV_PATH}' file failed validation!"
     echo "----------------------------------------------------"
     ret_code=1
 }
 
-while IFS= read -r LINE; do
-    echo "Checking format of: '${LINE}'"
-    [[ "${LINE}" = *[[:space:]]* ]] && {
-        echo "ERROR: Line contains white-space and it shouldn't!"
-        echo "--------------------------------------------------"
-        ret_code=1
-        continue
-    }
-    [[ "${LINE}" != *=* ]] && {
-        echo "ERROR: Line doesn't contain '=' and it should!"
-        echo "----------------------------------------------"
-        ret_code=1
-        continue
-    }
+process_file() {
+    while IFS= read -r LINE; do
+        echo "Checking format of: '${LINE}'"
+        [[ "${LINE}" = *[[:space:]]* ]] && {
+            echo "ERROR: Line contains white-space and it shouldn't!"
+            echo "--------------------------------------------------"
+            ret_code=1
+            continue
+        }
+        [[ "${LINE}" != *=* ]] && {
+            echo "ERROR: Line doesn't contain '=' and it should!"
+            echo "----------------------------------------------"
+            ret_code=1
+            continue
+        }
 
-    IMAGE_VARIABLE=$(echo "${LINE}" | cut --delimiter '=' --field 1)
-    IMAGE_URL=$(echo "${LINE}" | cut --delimiter '=' --field 2)
+        IMAGE_VARIABLE=$(echo "${LINE}" | cut --delimiter '=' --field 1)
+        IMAGE_URL=$(echo "${LINE}" | cut --delimiter '=' --field 2)
 
-    test -n "${IMAGE_VARIABLE}" || {
-        echo "ERROR: Couldn't parse image variable - got empty value!"
-        echo "-------------------------------------------------------"
-        ret_code=1
-        continue
-    }
+        test -n "${IMAGE_VARIABLE}" || {
+            echo "ERROR: Couldn't parse image variable - got empty value!"
+            echo "-------------------------------------------------------"
+            ret_code=1
+            continue
+        }
 
-    test -n "${IMAGE_URL}" || {
-        echo "ERROR: Couldn't parse image URL - got empty value!"
-        echo "--------------------------------------------------"
-        ret_code=1
-        continue
-    }
+        test -n "${IMAGE_URL}" || {
+            echo "ERROR: Couldn't parse image URL - got empty value!"
+            echo "--------------------------------------------------"
+            ret_code=1
+            continue
+        }
 
-    check_image "${IMAGE_VARIABLE}" "${IMAGE_URL}" || {
-        echo "ERROR: Image definition for '${IMAGE_VARIABLE}' isn't okay!"
-        echo "------------------------"
-        ret_code=1
-        continue
-    }
-done < "${PARAMS_ENV_PATH}"
+        check_image "${IMAGE_VARIABLE}" "${IMAGE_URL}" || {
+            echo "ERROR: Image definition for '${IMAGE_VARIABLE}' isn't okay!"
+            echo "------------------------"
+            ret_code=1
+            continue
+        }
+    done < "${1}"
+}
```

Comment:
we should return value of the `ret_code`  variable before exiting the function; also, it would be better to initialize the `ret_code` variable to 0 at the beginning of the function, just for the clarity, wdyt?

Reply:
Include the ret_code, as the ret_code was global var, 
didnt  return it from func, used it as it.
as the value gets updated one after another.

Reply:
Hm, okay. If we returned the `ret_code` value, then we could use e.g. this:
```if process_file "${file_}"; then
```

Reply:
Absolutely, i have included the new changes

Reply:
I would use `local ret_code=0` to keep the variable just for the local scope.

Anyway, I'm not gonna block this anymore. Thank you for your changes, Harshad!